### PR TITLE
Remove mhv-inherited-proofing

### PR DESCRIPTION
**Pull request by ianmagenta**
## Summary
Removed `mhv-inherited-proofing` (since it's dead code) and cleaned up parts of the site that referenced it.

## Related issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/74148

## What areas of the site does it impact?
This mainly impacts `mhv-inherited-proofing` and `site-wide/user-nav`.

## Acceptance criteria
- [x] Remove the `mhv-inherited-proofing` pages, components, and unit tests
- [x] Ensure the `user-nav/Main.jsx` is cleaned up and the modal's & Google Analytics tracking is removed
- [x] Ensure the `featureFlagNames.js` are removed

### CODE REVIEW

**Comment by Andrew565** [comment]
PR has failing required tests, opting not to review at this time. - Tier 2/FE COP

**Comment by ianmagenta** [comment]
@Andrew565 - All required tests are now passing, should be good for review!

File: src/platform/site-wide/user-nav/containers/Main.jsx [hunk 1]
```diff
@@ -287,14 +285,14 @@ export const mapStateToProps = state => {
     isLOA3: isLOA3(state),
     authenticatedWithOAuth: isAuthenticatedWithOAuth(state),
     isProfileLoading: isProfileLoading(state),
-    mhvTransition: mhvTransitionEnabled(state),
+    mhvTransition: false,
```
Comment (asg5704) [inline src/platform/site-wide/user-nav/containers/Main.jsx:288]: Remove this part

File: src/platform/site-wide/user-nav/containers/Main.jsx [hunk 2]
```diff
@@ -287,14 +285,14 @@ export const mapStateToProps = state => {
...
     mhvTransitionModal: mhvTransitionModalEnabled(state),
     signInServiceName: signInServiceNameSelector(state),
     shouldConfirmLeavingForm,
     useSignInService: signInServiceEnabled(state),
     user: selectUser(state),
     userGreeting: selectUserGreeting(state),
-    canTransferMHVAccount: transitionMHVAccount(state),
+    canTransferMHVAccount: false,
```
Comment (asg5704) [inline src/platform/site-wide/user-nav/containers/Main.jsx:295]: Remove the entire line

**Review by asg5704 — APPROVED** [review]

**Review by rmessina1010 — APPROVED** [review]
LGTM

**Pull request merged**